Assets/Scripts/UI/PauseMenuUI.cs

```diff
@@ -11,6 +11,8 @@ public class PauseMenuUI : MonoBehaviour
     private bool gameIsPaused = false;
     public KeyCode pauseKey;
 
+    private CursorLockMode prevCursorLockMode;
```

Comment:
you also need to keep track of the previous cursor visibility

Reply:
Fixed!